File: ios/UgoiraViewManager.m
#import <React/RCTViewManager.h>

@interface UgoiraViewManager : RCTViewManager
@end

@implementation UgoiraViewManager

RCT_EXPORT_MODULE()

RCT_EXPORT_METHOD(setResizeMode:(NSString *)resizeMode resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
    NSArray *acceptedValues = @[@"fitCenter", @"centerCrop", @"centerInside", @"contain", @"cover", @"stretch"];
    
    if (![acceptedValues containsObject:resizeMode]) {
        NSError *error = [NSError errorWithDomain:@"UgoiraViewManager" code:400 userInfo:@{NSLocalizedDescriptionKey: @"Invalid resizeMode value."}];
        reject(@"invalid_resize_mode", @"The provided resizeMode is not valid.", error);
        return;
    }
    
    // Handle the resizing behavior here
    // ...

    resolve(@{@"status": @"success", @"resizeMode": resizeMode});
}

@end

File: ios/UgoiraViewManager.swift
package com.yourpackage

import React
import UIKit

class UgoiraViewManager: RCTViewManager {
    
    override func getName() -> String {
        return "UgoiraView"
    }
    
    override func createViewInstance(_ context: RCTContext) -> UgoiraView {
        return UgoiraView()
    }
    
    @objc func setResizeMode(_ view: UgoiraView, resizeMode: String) {
        let acceptedValues = ["fitCenter", "centerCrop", "centerInside", "contain", "cover", "stretch"]
        if acceptedValues.contains(resizeMode) {
            view.resizeMode = resizeMode
        } else {
            view.resizeMode = "fitCenter"
        }
        DispatchQueue.main.async {
            view.setNeedsLayout()
        }
    }
}